frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/StaticNumberForm.tsx

```diff
@@ -0,0 +1,77 @@
+import type { FormEvent } from "react";
+import { useState } from "react";
+import { t } from "ttag";
+import {
+  Box,
+  Flex,
+  NumberInput,
+  PopoverBackButton,
+  TextInput,
+} from "metabase/ui";
+import type { SelectedComparisonStaticNumber } from "metabase-types/api";
+import { COMPARISON_TYPES } from "../constants";
+import { DoneButton } from "./SmartScalarSettingsWidgets.styled";
+
+interface StaticNumberFormProps {
+  value: Partial<SelectedComparisonStaticNumber>;
```

Comment:
```suggestion
  value?: SelectedComparisonStaticNumber;
```

We cannot say that the `selectedValue` that we are passing to this component is actually the `SelectedComparisonStaticNumber`. If we want to, we can optionally pass the selectedValue if we know it is of type `staticNumber` just like we do in the `PeriodsAgoOptionComponent`.

```
selectedValue={selectedValue.type === type ? selectedValue : undefined}
```

Then we will know for sure we have `undefined | SelectedComparisonStaticNumber`